Add ResNet to models exportable with ONNX

chamidullinr opened the pull request:
# What does this PR do?

<!--
Congratulations! You've made it this far! You're not quite done yet though.

Once merged, your PR is going to appear in the release notes with the title you set, so make sure it's a great title that fully reflects the extent of your awesome contribution.

Then, please replace this with a description of the change and which issue is fixed (if applicable). Please also include relevant motivation and context. List any dependencies (if any) that are required for this change.

Once you're done, someone will review your PR shortly (see the section "Who can review?" below to tag some potential reviewers). They may suggest changes to make the code even better. If no one reviewed your PR after a week has passed, don't hesitate to post a new comment @-mentioning the same persons---sometimes notifications get lost.
-->

<!-- Remove if not applicable -->

I added OnnxConfig to make ResNet model available for ONNX conversion.

Issue [#16308](https://github.com/huggingface/transformers/issues/16308)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?


## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

<!-- Your PR will be replied to more quickly if you can figure out the right person to tag with @

 If you know how to use git blame, that is the easiest way, otherwise, here is a rough guide of **who to tag**.
 Please tag fewer than 3 people.

Models:

- albert, bert, xlm: @LysandreJik
- blenderbot, bart, marian, pegasus, encoderdecoder,  t5: @patrickvonplaten, @patil-suraj
- longformer, reformer, transfoxl, xlnet: @patrickvonplaten
- fsmt: @stas00
- funnel: @sgugger
- gpt2: @patrickvonplaten, @LysandreJik
- rag: @patrickvonplaten, @lhoestq
- tensorflow: @LysandreJik

Library:

- benchmarks: @patrickvonplaten
- deepspeed: @stas00
- ray/raytune: @richardliaw, @amogkam
- text generation: @patrickvonplaten
- tokenizers: @n1t0, @LysandreJik
- trainer: @sgugger
- pipelines: @LysandreJik

Documentation: @sgugger

HF projects:

- datasets: [different repo](https://github.com/huggingface/datasets)
- rust tokenizers: [different repo](https://github.com/huggingface/tokenizers)

Examples:

- maintained examples (not research project or legacy): @sgugger, @patil-suraj
- research_projects/bert-loses-patience: @JetRunner
- research_projects/distillation: @VictorSanh

 -->

chamidullinr committed:
Add ResNet to models exportable with ONNX

Commit:
Add onnx config for RoFormer (#16861)

* add roformer onnx config

HuggingFaceDocBuilderDev commented:
The docs for this PR live [here](https://moon-ci-docs.huggingface.co/docs/transformers/pr_16948). All of your documentation changes will be reflected on that endpoint.

mingboiz committed:
documentation: some minor clean up (#16850)

ftnext committed:
Fix RuntimeError message format (#16906)

tricktreat committed:
use original loaded keys to find mismatched keys (#16920)

anton-l committed:
[Research] Speed up evaluation for XTREME-S (#16785)

* Avoid repeated per-lang filtering

* Language groups and logits preprocessing

* Style

NielsRogge committed:
Add semantic script, trainer (#16834)

* Add first draft

* Improve script and README

* Improve README

* Apply suggestions from code review

* Improve script, add link to resulting model

* Add corresponding test

* Adjust learning rate

ydshieh committed:
Fix HubertRobustTest PT/TF equivalence test on GPU (#16943)

Co-authored-by: ydshieh <ydshieh@users.noreply.github.com>

ChainYo commented on src/transformers/models/resnet/configuration_resnet.py:
```diff
@@ -89,3 +95,20 @@ def __init__(
+
+    @property
+    def atol_for_validation(self) -> float:
+        return 1e-4
```
Hey @chamidullinr nice PR, did you try the conversion config with the default `--atol=1e-5` before putting it to `1e-4` ?

If you are okay you can push a ResNet converted model to the [ONNXConfig for all](https://huggingface.co/OWG) organisation, it would be awesome! :hugs:

chamidullinr replied:
I pushed a `ResNet-50` ONNX converted model. And I set `--atol=1e-3` because some tests were failing.

lewtun reviewed (approved):
Thanks for adding the first non-Transformer model to our ONNX exporter @chamidullinr !

Apart from @ChainYo's comment, this all looks good to me, so can you please check that the slow tests pass by running:

```
RUN_SLOW=1 pytest tests/onnx/test_onnx_v2.py -s -k "resnet"
```

Gently pinging @sgugger for final approval 🤗

sgugger reviewed (approved):
Thanks for the PR. LGTM module existing comments :-)

searchivarius committed:
Misc. fixes for Pytorch QA examples: (#16958)

1. Fixes evaluation errors popping up when you train/eval on squad v2 (one was newly encountered and one that was previously reported Running SQuAD 1.0 sample command raises IndexError #15401 but not completely fixed).
2. Removes boolean arguments that don't use store_true. Please, don't use these: *ANY non-empty string is being converted to True in this case and this clearly is not the desired behavior (and it creates a LOT of confusion).
3. All no-trainer test scripts are now saving metric values in the same way (with the right prefix eval_), which is consistent with the trainer-based versions.
4. Adds forgotten model.eval() in the no-trainer versions. This improved some results, but not everything (see the discussion in the end). Please, see the F1 scores and the discussion below.

NielsRogge committed:
[HF Argparser] Fix parsing of optional boolean arguments (#16946)

* Add fix

* Apply suggestion from code review

Co-authored-by: Niels Rogge <nielsrogge@Nielss-MacBook-Pro.local>

Yard1 committed:
Fix `distributed_concat` with scalar tensor (#16963)

* Fix `distributed_concat` with scalar tensor

* Update trainer_pt_utils.py

mishig25 committed:
Update custom_models.mdx (#16964)

BertModelForSequenceClassification -> BertForSequenceClassification

chamidullinr commented:
Thanks for the suggestions. One test fails when running slow tests. The error is:
> ValueError: Outputs values doesn't match between reference model and ONNX exported model: Got max absolute difference of: 0.000244140625

Should I set --atol=1e-3 or is there some way to fix this?

sgugger committed:
Fix add-new-model-like when model doesn't support all frameworks (#16966)

sgugger committed:
Fix multiple deletions of the same files in save_pretrained (#16947)

* Fix multiple deletions of the same files in save_pretrained

* Add is_main_process argument

lewtun commented:
> Should I set --atol=1e-3 or is there some way to fix this?

Ah yes, some models require a lower tolerance due to their architectures. Setting `--atol=1e-3` as the default is fine!

muellerzr committed:
Fixup no_trainer save logic (#16968)

* Fixup all examples

sgugger committed:
Fix doc notebooks links (#16969)

* Fix doc notebooks links

* Remove missing section

ydshieh committed:
Fix check_all_models_are_tested (#16970)

Co-authored-by: ydshieh <ydshieh@users.noreply.github.com>

ydshieh committed:
Add -e flag to some GH workflow yml files (#16959)

* Add -e flag

* add check

* create new keys

* run python setup.py build install

* add comments

* change to develop

Co-authored-by: ydshieh <ydshieh@users.noreply.github.com>

datquocnguyen committed:
Update tokenization_bertweet.py (#16941)

The emoji version must be either 0.5.4 or 0.6.0. Newer emoji versions have been updated to newer versions of the Emoji Charts, thus not consistent with the one used for pre-processing the pre-training Tweet corpus (i.e. not consistent with the vocab).

chamidullinr committed:
Changed atol in ResNet ONNX configuration

ChainYo commented:
Can't we rebase this branch and add the code for ResNet ?

regisss commented:
I'm taking care of this in #17585, I didn't see this PR before opening mine. @ChainYo

It should be done by the end of the week!

ChainYo commented:
> It should be done by the end of the week!

Pretty cool!
